galley/pkg/runtime/projections/serviceentry/converter/instance.go

```diff
@@ -65,14 +65,19 @@ func (i *Instance) convertService(service *resource.Entry, outMeta *mcp.Metadata
 	}
 
 	spec := service.Item.(*coreV1.ServiceSpec)
-
-	resolution := networking.ServiceEntry_STATIC
 	location := networking.ServiceEntry_MESH_INTERNAL
 	endpoints := convertExternalServiceEndpoints(spec, service.Metadata)
+	var resolution networking.ServiceEntry_Resolution
+	// Resolution STATIC must have endpoints
+	if len(endpoints) != 0 {
+		resolution = networking.ServiceEntry_STATIC
+	}
 
 	// Check for an external service
+	host := serviceHostname(service.ID.FullName, i.domain)
 	externalName := ""
-	if spec.Type == coreV1.ServiceTypeExternalName && spec.ExternalName != "" {
+	err := model.ValidateFQDN(host)
+	if err == nil && spec.Type == coreV1.ServiceTypeExternalName && spec.ExternalName != "" {
```

Comment:
why is the validation result of the hostname included in this `if` condition?

Reply:
This is to distinguish between the STATIC and DNS. DNS resolution requires a valid FQDN.

Reply:
Sure, SGTM